src/cryptography/x509.py

```diff
@@ -249,6 +264,96 @@ def __repr__(self):
                 "path_length={0.path_length})>").format(self)
 
 
+class GeneralName(object):
+    _gn_str_types = [
+        GeneralNameType.rfc822_name,
+        GeneralNameType.dns_name,
+        GeneralNameType.uniform_resource_identifier,
+    ]
+
+    def __init__(self, type, value):
+        if not isinstance(type, GeneralNameType):
+            raise TypeError("type must be a GeneralNameType")
+
+        if type in [
+            GeneralNameType.other_name,
+            GeneralNameType.x400_address,
+            GeneralNameType.edi_party_name
+        ]:
+            raise ValueError("{0} is not a supported type".format(type))
+
+        if (
+            type in self._gn_str_types and not
+            isinstance(value, six.string_types)
```

Comment:
IDNA domains are encoded in the certificate as that strict subset. For example, Unicode biohazard (or radiation I can never remember) is xn--k4h. That said, we could potentially choose to parse that data and return the Unicode rather than the decomposed version that's actually present in the certificate... Is there a decent IDNA parser library for Python?



> On Mar 31, 2015, at 8:41 PM, Alex Gaynor <notifications@github.com> wrote:
> 
> In src/cryptography/x509.py:
> 
> > +    ]
> > +
> > +    def __init__(self, type, value):
> > +        if not isinstance(type, GeneralNameType):
> > +            raise TypeError("type must be a GeneralNameType")
> > +
> > +        if type in [
> > +            GeneralNameType.other_name,
> > +            GeneralNameType.x400_address,
> > +            GeneralNameType.edi_party_name
> > +        ]:
> > +            raise ValueError("{0} is not a supported type".format(type))
> > +
> > +        if (
> > +            type in self._gn_str_types and not
> > +            isinstance(value, six.string_types)
> Are they a strict subset of ascii? What format are they in here? Unicode domain names are definitely a thing, see punycode and such.
> …
> —
> Reply to this email directly or view it on GitHub.
>